homeassistant/components/airq/sensor.py

```diff
@@ -0,0 +1,205 @@
+
+    coordinator = hass.data[DOMAIN][config.entry_id]
+    available_keys = list(coordinator.data.keys())
+
+    # Add sensors under warmup
+    status = coordinator.data["Status"]
+    if isinstance(status, dict):
+        warming_up_sensors = [
+            k for k, v in status.items() if "sensor still in warm up phase" in v
+        ]
+        available_keys.extend(warming_up_sensors)
+        _LOGGER.debug(
+            "Following %d sensors are warming up: %s",
+            len(warming_up_sensors),
+            ", ".join(warming_up_sensors),
+        )
+
+    # Filter out non-sensor keys and build a list of SensorEntityDescription objects
+    available_sensors = [
+        description for description in SENSOR_TYPES if description.key in available_keys
+    ]
+    _LOGGER.debug(
+        "Identified %d  available sensors: %s",
+        len(available_sensors),
+        ", ".join([sensor.key for sensor in available_sensors]),
+    )
+
+    entities = [
+        AirQSensor(coordinator, description) for description in available_sensors
+    ]
+    async_add_entities(entities)
+
+
+class AirQSensor(CoordinatorEntity, SensorEntity):
+    """Representation of a Sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: AirQCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize a single sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+
+        self._attr_device_info = DeviceInfo(
+            # the name (e.g. ABCDE) will be prepended to description.name: 'ABCDE NO2'
+            name=coordinator.config["name"],
+            model=coordinator.config["model"],
+            sw_version=coordinator.config["sw_version"],
+            hw_version=coordinator.config["hw_version"],
+            identifiers={(DOMAIN, coordinator.config["id"])},
+            manufacturer=MANUFACTURER,
+            suggested_area=coordinator.config["room_type"],
+        )
+        self._attr_name = description.name
+        self._attr_unique_id = f"{coordinator.config['id']}_{description.key}"
+
+    @property
+    def native_value(self) -> float | int | None:
+        """Return the value reported by the sensor."""
+        # airthings has a neat way of doing it when the data returned by the API
+        # are a dictionary with keys for each device, and values being dictionaries
+        # of sensor values. Under this condition, the call should be:
+        # return self.coordinator.data[self._id].sensors[self.entity_description.key]
+        # In our case now, only one device is allowed and self.coordinator.data
+        # contains the regular dict retrieved from a single device
+
+        # .get(key, None) over [key] to guard against the keys of the warming up
+        # sensors not being present in the returned dictionary
+        return self.coordinator.data.get(self.entity_description.key, None)
```

Comment:
`get()` returns default value `None` if key is not available.

Reply:
Thanks for the hint!